src/grad.jl

```diff
@@ -27,28 +27,28 @@ end
 Approximate the Jacobian of `f` at `x` using `fdm`. `f(x)` must be a length `D` vector. If
 `D` is not provided, then `f(x)` is computed once to determine the output size.
 """
-function jacobian(fdm, f, x::Vector{T}, D::Int) where {T<:Real}
+function jacobian(fdm, f, x::Vector{T}, D::Int) where {T<:Number}
     J = Matrix{T}(undef, D, length(x))
     for d in 1:D
         J[d, :] = grad(fdm, x->f(x)[d], x)
     end
     return J
 end
-jacobian(fdm, f, x::Vector{<:Real}) = jacobian(fdm, f, x, length(f(x)))
+jacobian(fdm, f, x::Vector{<:Number}) = jacobian(fdm, f, x, length(f(x)))
 
 """
     _jvp(fdm, f, x::Vector{<:Real}, ẋ::AbstractVector{<:Real})
 
 Convenience function to compute `jacobian(f, x) * ẋ`.
 """
-_jvp(fdm, f, x::Vector{<:Real}, ẋ::AV{<:Real}) = fdm(ε -> f(x .+ ε .* ẋ), zero(eltype(x)))
+_jvp(fdm, f, x::Vector{<:Number}, ẋ::AV{<:Number}) = fdm(ε -> f(x .+ ε .* ẋ), zero(eltype(x)))
 
 """
     _j′vp(fdm, f, ȳ::AbstractVector{<:Real}, x::Vector{<:Real})
 
 Convenience function to compute `jacobian(f, x)' * ȳ`.
 """
-_j′vp(fdm, f, ȳ::AV{<:Real}, x::Vector{<:Real}) = jacobian(fdm, f, x, length(ȳ))' * ȳ
+_j′vp(fdm, f, ȳ::AV{<:Number}, x::Vector{<:Number}) = transpose(jacobian(fdm, f, x, length(ȳ))) * ȳ
```

Comment:
Can the `Real` -> `Number` change be done for the entire file? Currently `Real` will remain in a few places

Reply:
For now, I only changed the constraints, where I actually needed too. I think it would be safer for now to leave the other constraints and only loosen them when needed in ChainRules.jl

Reply:
Hmm, yeah, I do see that point of view

It's important to make sure FiniteDifferences itself has consistent behaviour e.g. it is surprising for `to_vec` to work on a Complex number but not an array of Complex numbers

But 🤷‍♂ let's see what Will or Wessel think

Reply:
`to_vec` seems harmless enough to correctly handle `Vector{<:Number}`, so I've added that definition.

Reply:
I also suppose, it would make more sense to rename `j′vp` to `jᵀvp`, but that's would definitely be breaking and out of the scope of this PR